library/src/main/java/kwasm/format/text/Type.kt

```diff
@@ -14,37 +14,54 @@
 
 package kwasm.format.text
 
+import kwasm.ast.ValueType
 import kwasm.format.ParseContext
+import kwasm.format.ParseException
 
-sealed class Type(
+sealed class Type<T>(
     protected val sequence: CharSequence,
     protected val context: ParseContext? = null
 ) {
+    val value: T by lazy {
+        parseValue()
+    }
 
-    protected abstract fun parseValue(): Unit
+    protected abstract fun parseValue(): T
 
     class ValueType(
         sequence: CharSequence,
         context: ParseContext? = null
-    ) : Type(sequence, context) {
-        override fun parseValue() {
-            TODO("not implemented")
+    ) : Type<kwasm.ast.ValueType>(sequence, context) {
+        override fun parseValue(): kwasm.ast.ValueType {
+            return when (sequence){
+                "i32" -> kwasm.ast.ValueType.I32
+                "i64" -> kwasm.ast.ValueType.I64
+                "f32" -> kwasm.ast.ValueType.F32
+                "f64" -> kwasm.ast.ValueType.F64
+                else -> {
+                    throw ParseException("Invalid ValueType", context)
+                }
+            }
```

Comment:
Nit: this could be tidied up a little:

```kotlin
override fun parseValue(): kwasm.ast.ValueType = when (sequence) {
    "i32" -> kwasm.ast.ValueType.I32,
    // the other options
    else -> throw ParseException("Invalid ValueType", context)
}
```

Reply:
Done